core/CronArchive.php

```diff
@@ -1003,36 +1014,59 @@ private function isThereAValidArchiveForPeriod($idSite, $period, $date, $segment
         // if there is an invalidated archive within the range, find out the oldest one and how far it is from today,
         // and change the lastN $date to be value so it is correctly re-processed.
         $newDate = $date;
-        if (!$isThereArchiveForAllPeriods
-            && preg_match('/^last([0-9]+)/', $date, $matches)
-        ) {
-            $lastNValue = (int) $matches[1];
-
-            usort($diff, function ($lhs, $rhs) {
-                $lhsDate = explode(',', $lhs)[0];
-                $rhsDate = explode(',', $rhs)[0];
-
-                if ($lhsDate == $rhsDate) {
-                    return 1;
-                } else if (Date::factory($lhsDate)->isEarlier(Date::factory($rhsDate))) {
-                    return -1;
-                } else {
-                    return 1;
-                }
-            });
+        if ($isLast) {
+            if (!$isThereArchiveForAllPeriods) {
+                $lastNValue = (int)$matches[1];
+
+                usort($diff, function ($lhs, $rhs) {
+                    $lhsDate = explode(',', $lhs)[0];
+                    $rhsDate = explode(',', $rhs)[0];
+
+                    if ($lhsDate == $rhsDate) {
+                        return 1;
+                    } else if (Date::factory($lhsDate)->isEarlier(Date::factory($rhsDate))) {
+                        return -1;
+                    } else {
+                        return 1;
+                    }
+                });
 
-            $oldestDateWithoutArchive = explode(',', reset($diff))[0];
-            $todayInTimezone = Date::factoryInTimezone('today', Site::getTimezoneFor($idSite));
+                $oldestDateWithoutArchive = explode(',', reset($diff))[0];
+                $todayInTimezone = Date::factoryInTimezone('today', Site::getTimezoneFor($idSite));
 
-            /** @var Range $newRangePeriod */
-            $newRangePeriod = PeriodFactory::build($period, $oldestDateWithoutArchive . ',' . $todayInTimezone);
+                /** @var Range $newRangePeriod */
+                $newRangePeriod = PeriodFactory::build($period, $oldestDateWithoutArchive . ',' . $todayInTimezone);
 
-            $newDate = 'last' . min($lastNValue, $newRangePeriod->getNumberOfSubperiods());
+                $newDate = 'last' . min($lastNValue, $newRangePeriod->getNumberOfSubperiods());
+            } else if ($isTodayIncluded) {
+                $isThereArchiveForAllPeriods = false;
+                $newDate = 'last1';
+            }
```

Comment:
btw... I was thinking... if it is `last1` (aka `today`, `this week`, `this month`), then we could technically get the date when this archive was last generated, and if it was generated less than `time_before_(today|$period)_archive_considered_outdated` seconds ago, return `$isThereArchiveForAllPeriods  = true` and skip the archive this time. 

Only if it's easy to do though. Otherwise could also just launch the archiving. On cloud where we consider archives outdated sometimes only evert 6, 12, 18 hours (depending on config), then it could save us launching quite a few archive launches.

Not sure it's clear what I mean? It be probably similar to what we have in archive launcher.

Reply:
I think this is already done, regardless of whether browser archiving is used or not. In Loader.php we always use the result of `getMinTimeArchiveProcessed()`. So it should always look for an archive that is out of date, correct? I could maybe add a debug log for the future...